src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py

```diff
@@ -663,5 +664,28 @@ def on_goto_request(self, py_db, request):
         response = pydevd_base_schema.build_response(request, kwargs={'body': {}})
         return NetCommand(CMD_RETURN, 0, response, is_json=True)
 
+    def on_setdebuggerproperty_request(self, py_db, request):
+        args = request.arguments.kwargs
+        if 'dontTraceStartPatterns' in args and 'dontTraceEndPatterns' in args:
+            start_patterns = tuple(args['dontTraceStartPatterns'])
+            end_patterns = tuple(args['dontTraceEndPatterns'])
+
+            def dont_trace_files(abs_path):
```

Comment:
We should check that after we receive the `dontTraceStartPatterns` / `dontTraceEndPatterns`, if it's called again it's always called with the same parameters and that `get_file_type` hasn't been called before this request (check if something is already cached) as `get_file_type` is not expected to change its return value (if that's needed, more work is needed in the debugger as other caches rely on this).

You can do this all local to this function (maybe rename `dont_trace_files` to `dont_trace_files_property_request` and check the function name and add the `start_patterns` and the `end_patterns` to the function itself to see if they're still equal and show a critical warning in stderr and return a failure response if the invariant is broken -- please create a test case for this specific situation).